### PR TITLE
Acwd1m pacakging

### PR DESCRIPTION
**Pull request by Sckorn**
Redone packaging of old packages. Packed newly introduced ones.

### CODE REVIEW

File: common/BUILD [hunk 1]
```diff
@@ -131,14 +135,320 @@ pkg_tar(
...
+
+# filesystem-dev
+
+pkg_tar(
+    name = "filesystem-data-headers",
+    srcs = [
+        "include/common/utils/file_utils.h",
+    ],
+    mode = "0644",
+    package_dir = "/usr/local/include/rcbe/utils",
```
Comment (radameon) [inline common/BUILD:222]: `/usr/local/include/rcbe/common/utils`

File: common/BUILD [hunk 2]
```diff
@@ -131,14 +135,320 @@ pkg_tar(
...
+    ],
+)
+
+pkg_deb(
+    name = "common-dev-deb",
+    architecture = TARGET_ARCH,
+    built_using = "unzip (6.0.1)",
+    data = ":common-filesystem-dev-package-data",
+    depends = ["rcbe_common_utils_filesystem_{}_{}".format(ENGINE_VERSION, TARGET_ARCH)],
+    description = "RCBE game engine common functionality headers package",
```
Comment (radameon) [inline common/BUILD:240]: `... common filesystem functionality headers ...`

File: common/BUILD [hunk 3]
```diff
@@ -131,14 +135,320 @@ pkg_tar(
...
+
+# json-dev
+
+pkg_tar(
+    name = "json-data-headers",
+    srcs = [
+        "include/common/utils/json_utils.hpp",
+    ],
+    mode = "0644",
+    package_dir = "/usr/local/include/rcbe/utils",
```
Comment (radameon) [inline common/BUILD:255]: `/usr/local/include/rcbe/common/utils`

File: common/BUILD [hunk 4]
```diff
@@ -131,14 +135,320 @@ pkg_tar(
...
+
+# fundamentals-dev
+
+pkg_tar(
+    name = "fundamentals-data-headers",
+    srcs = glob([
+        "include/common/fundamentals/**/*.hpp",
+    ]),
+    mode = "0644",
+    package_dir = "/usr/local/include/rcbe/fundamentals",
```
Comment (radameon) [inline common/BUILD:290]: Again missing `common` in `package_dir`

File: common/BUILD [hunk 5]
```diff
@@ -131,14 +135,320 @@ pkg_tar(
...
+
+# math-utils-dev
+
+pkg_tar(
+    name = "math-utils-data-headers",
+    srcs = glob([
+        "include/common/utils/math_utils.hpp",
+    ]),
+    mode = "0644",
+    package_dir = "/usr/local/include/rcbe/utils",
```
Comment (radameon) [inline common/BUILD:322]: `common` is missing

File: common/BUILD [hunk 6]
```diff
@@ -131,14 +135,320 @@ pkg_tar(
...
+
+# output-utils-dev
+
+pkg_tar(
+    name = "output-utils-data-headers",
+    srcs = glob([
+        "include/common/utils/output_utils.hpp",
+    ]),
+    mode = "0644",
+    package_dir = "/usr/local/include/rcbe/utils",
```
Comment (radameon) [inline common/BUILD:355]: `common`

File: common/BUILD [hunk 7]
```diff
@@ -131,14 +135,320 @@ pkg_tar(
...
+
+# fuzzy-logic-dev
+
+pkg_tar(
+    name = "fuzzy-logic-data-headers",
+    srcs = glob([
+        "include/common/fuzzy_logic/fuzzy_logic.hpp",
+    ]),
+    mode = "0644",
+    package_dir = "/usr/local/include/rcbe/fuzzy_logic",
```
Comment (radameon) [inline common/BUILD:392]: `common`

File: common/BUILD [hunk 8]
```diff
@@ -131,14 +135,320 @@ pkg_tar(
...
+# delegate-dev
+
+pkg_tar(
+    name = "delegate-data-headers",
+    srcs = glob([
+        "include/common/delegate/AbstractDelegate.hpp",
+        "include/common/delegate/Delegate.hpp",
+    ]),
+    mode = "0644",
+    package_dir = "/usr/local/include/rcbe/delegate",
```
Comment (radameon) [inline common/BUILD:428]: `common`

File: datamodel/BUILD [hunk 9]
```diff
@@ -157,43 +157,485 @@ cc_test(
...
+        ":datamodel-core-data",
+    ],
+)
+
+pkg_deb(
+    name = "datamodel-core-deb",
+    architecture = TARGET_ARCH,
+    built_using = "unzip (6.0.1)",
+    data = ":datamodel-core-package-data",
+    #depends = ["rcbe_common_fundamentals_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH)],
```
Comment (radameon) [inline datamodel/BUILD:194]: remove commented out code

File: deploy_packages.py [hunk 10]
```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/python
+
+import sys
+import os
+import shutil
+from shutil import copyfile
+
+deployable_packages = ['common', 'datamodel', 'parsers', 'gl_extensions', 'input_manager', 'renderer', 'ticker', 'traits', 'window_manager', 'camera'] # left for later use
```
Comment (radameon) [inline deploy_packages.py:8]: This should probably be read from a conf file anyway, so I see little reason to leave it be

File: deploy_packages.py [hunk 11]
```diff
@@ -0,0 +1,107 @@
...
+
+def usage():
```
Comment (radameon) [inline deploy_packages.py:10]: Add a TODO to make a proper program options parse, with proper description and usage.

File: deploy_packages.py [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+            elif os.path.isdir(file_path):
+                shutil.rmtree(file_path)
+        except Exception as e:
+            print('Failed to delete %s. Reason: %s' % (file_path, e))
+
+def clear_and_rm_dir(tmp_dir):
+    try:
+        clear_dir(tmp_dir)
+        os.rmdir(tmp_dir)
+    except OSError:
```
Comment (radameon) [inline deploy_packages.py:31]: Consider not catching it here and let the caller decide

File: meta_package/BUILD [hunk 13]
```diff
@@ -7,8 +7,8 @@ pkg_tar(
     name = "all-package-data",
     extension = "tar.gz",
     deps = [
```
Comment (radameon) [inline meta_package/BUILD:9]: This whole metapacking stuff is no longer needed remove the code, but leave the `BUILD` file be.

File: parsers/BUILD [hunk 14]
```diff
@@ -58,16 +64,54 @@ pkg_tar(
...
+
+pkg_deb(
+    name = "parsers-headers-deb",
+    architecture = TARGET_ARCH,
+    built_using = "unzip (6.0.1)",
+    data = ":parsers-headers-package-data",
+    depends = [
+        "rcbe_common_filesystem_utils_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
+        "rcbe_datamodel_geometry_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
+        "rcbe_datamodel_math_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
```
Comment (radameon) [inline parsers/BUILD:110]: Should depend on `rcbe_parsers`

File: renderer/BUILD [hunk 15]
```diff
@@ -78,18 +82,65 @@ pkg_tar(
...
+    built_using = "unzip (6.0.1)",
+    data = ":renderer-headers-package-data",
+    depends = [
+        "rcbe_datamodel_core_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
+        "rcbe_datamodel_math_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
+        "rcbe_datamodel_rendering_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
+        "rcbe_datamodel_visual_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
+        "rcbe_gl_extensions_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
+        "rcbe_window_manager_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
+        "rcbe_ticker__dev{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
```
Comment (radameon) [inline renderer/BUILD:137]: Extra `_` symbol

File: renderer/BUILD [hunk 16]
```diff
@@ -78,18 +82,65 @@ pkg_tar(
...
+        "libboost-log{}-dev".format(BOOST_TARGET_VERSION),
+        "nlohmann-json3-dev",
```
Comment (radameon) [inline renderer/BUILD:139]: Should also depend on `rcbe_renderer`

File: window_manager/BUILD [hunk 17]
```diff
@@ -69,13 +73,63 @@ pkg_tar(
...
+    ],
+)
+
+pkg_deb(
+    name = "window-manager-headers-deb",
+    architecture = TARGET_ARCH,
+    built_using = "unzip (6.0.1)",
+    data = ":wm-headers-package-data",
+    depends = [
+        "rcbe_datamodel_core_dev_{}_{}".format(ENGINE_VERSION, TARGET_ARCH),
```
Comment (radameon) [inline window_manager/BUILD:121]: Must depend on `rcbe_window_manager`

**Review by radameon — CHANGES_REQUESTED** [review]
Fix these and release is good to go

File: deploy_packages.py (hunk 12 above)
Comment (Sckorn) [reply deploy_packages.py:31]: No need to do this, if it can't clean up so be it. It shouldn't crash the deploy. The files are stored inside `/tmp` anyway

Comment (radameon) [reply deploy_packages.py:30]: 'Kay

File: parsers/BUILD (hunk 14 above)
Comment (radameon) [reply parsers/BUILD:110]: See it below, good.

File: renderer/BUILD (hunk 16 above)
Comment (radameon) [reply renderer/BUILD:140]: See it above. Good

File: window_manager/BUILD (hunk 17 above)
Comment (radameon) [reply window_manager/BUILD:121]: See it below. Good

**Review by radameon — APPROVED** [review]
All good. Use that fancy deploy script, rename the output archive. And smack that tag after merge. Oh and don't forget to squash beforehead

**Pull request merged**